default/python/AI/InvasionAI.py

```diff
@@ -543,27 +527,45 @@ def evaluate_invasion_planet(planet_id):
     if clear_path:
         planet_score *= 1.5
     debug(
-        " - planet score: %.2f\n"
-        " - planned troops: %.2f\n"
-        " - projected troop cost: %.1f\n"
-        " - threat factor: %s\n"
-        " - planet detail: %s\n"
-        " - popval: %.1f\n"
-        " - bldval: %s\n"
-        " - enemyval: %s",
-        planet_score,
-        planned_troops,
-        troop_cost,
-        threat_factor,
-        detail,
-        colony_base_value,
-        bld_tally,
-        enemy_val,
+        f" - planet score: {planet_score:.2f}\n"
+        f" - planned troops: {planned_troops:.2f}\n"
+        f" - projected troop cost: {troop_cost:.1f}\n"
+        f" - threat factor: {threat_factor}\n"
+        f" - planet detail: {detail}\n"
+        f" - popval: {colony_base_value:.1f}\n"
+        f" - bldval: {bld_tally}\n"
+        f" - enemyval: {enemy_val}\n"
+        f" - system secured: {system_secured}"
     )
-    debug(" - system secured: %s" % system_secured)
     return [planet_score, planned_troops]
 
 
+def secure_system(system_id: SystemId, secure_now: bool):
+    """Check whether we have any military fleets in the system.
+    If secure_now is True, set one of the fleets to MissionType.SECURE
+    If secure_now is False, return true if there is any fleet that secure it.
+    """
+    universe = fo.getUniverse()
+    ai_state = get_aistate()
+    system = universe.getSystem(system_id)
+    secure_targets = [system_id] + list(system.planetIDs)
+    system_status = ai_state.systemStatus.get(system_id, {})
+    mission_types = [MissionType.SECURE, MissionType.MILITARY, MissionType.PROTECT_REGION, MissionType.ORBITAL_DEFENSE]
+    missions = ai_state.get_fleet_missions_with_any_mission_types(mission_types)
+    secured = False
+    for mission in missions:
+        fleet = universe.getFleet(mission.fleet.id)
+        if not fleet or fleet.systemID != system_id:
+            continue
+        target_obj = mission.target.get_object()
+        if target_obj is not None and target_obj.id in secure_targets:
+            if secure_now and mission.type != MissionType.SECURE:
+                mission.set_target(MissionType.SECURE, mission.target)
+            secured = system_status.get("myFleetRating", 0) > 0
+            break
+    return secured
```

Comment:
We could try to remove the `secured` variable. 

```suggestion
            return secured
    return False
```

Should we stop on the first fleet to we should keep trying?
If this felt is not

Reply:
I know that could isn't perfect. I copied the "myFleetRating" stuff. It seems to work in by and large...
I think I should rather check whether the fleet I just set to secure actually does the job, but I have to figure out how to do that. I'll add a TBD to make that clearer.